test/metabase/lib/convert_test.cljc

```diff
@@ -197,6 +197,32 @@
                          :limit        1
                          :source-table 4}})))))
 
+(deftest ^:parallel parameters-in-source-query-test
+  (is (=? {:lib/type :mbql/query
+           :stages
+           [{:lib/type :mbql.stage/mbql
+             :source-table 1
+             :parameters
+             [{:target
+               [:dimension
+                {:stage-number 0}
+                [:expression {} "customColumn"]]}]
+             :expressions
+             [[:field
+               {:base-type :type/BigInteger
+                :lib/expression-name "customColumn"}
+               2]]}]
+           :database 5
+           :lib.convert/converted? true}
+         (lib.convert/->pMBQL {:type :query
+                               :database 5
+                               :query {:source-table 1
+                                       :parameters [{:target [:dimension
+                                                              [:expression "customColumn"
+                                                               {:base-type :type/BigInteger}]
+                                                              {:stage-number 0}]}]
+                                       :expressions {"customColumn" [:field 2 {:base-type :type/BigInteger}]}}}))))
```

Comment:
Please use `lib.tu.macros/mbql-query` for this. It will help make the test case more future-proof (eg. by automatically adding `:expression-idents`, which you're missing).